File: Architecture.Web/ClientApp/src/app/Shared/Entity/Accounts/transactionRequest.ts
import { Auditable } from '../Common/auditable';
import { PaymentType } from '../LU/paymentType';
import { QueryObject } from "../Common/query-object";

export class TransactionRequest extends Auditable {
    public transactionRequestId: number;
    public purpose: string;
    public approvedBy: string;
    public approvedDate: Date;
    public requestBy: string;
    public transactionId: number;
    public paymentTypeId: number;
    public paymentType: PaymentType;
    public paymentReceivedBy: string;
    public paymentReceivedDate: Date;
    public amount: number = 0.0;
}

export class TransactionRequestQuery extends QueryObject {
    name: string;

    constructor(init?: Partial<TransactionRequestQuery>) {
        super();
        Object.assign(this, init);
    }

    clear() {
        this.name = '';
    }
}

File: Architecture.Core/Entities/Accounts/TransactionRequest.cs
using Architecture.Core.Entities.Core;
using Architecture.Core.Entities.LU;
using System;

namespace Architecture.Core.Entities.Accounts
{
    public class TransactionRequest : Auditable
    {
        public int TransactionRequestId { get; set; }
        public string Purpose { get; set; }
        public Guid? ApprovedBy { get; set; }
        public DateTime? ApprovedDate { get; set; }
        public Guid? RequestBy { get; set; }
        public int? TransactionId { get; set; }
        public int? PaymentTypeId { get; set; }
        public virtual PaymentType PaymentType { get;set;}
        public string PaymentReceivedBy { get; set; }
        public DateTime? PaymentReceivedDate { get; set; }
        public Decimal Amount { get; set; }
    }
}
